src/cascadia/TerminalSettingsModel/TerminalSettings.h

```diff
@@ -0,0 +1,221 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- TerminalSettings.h
+
+Abstract:
+- The implementation of the TerminalSettings winrt class. Provides both
+        terminal control settings and terminal core settings.
+Author(s):
+- Mike Griese - March 2019
+
+--*/
+#pragma once
+
+#include "TerminalSettings.g.h"
+#include "IInheritable.h"
+#include "../inc/cppwinrt_utils.h"
+#include <DefaultSettings.h>
+#include <conattrs.hpp>
+
+// fwdecl unittest classes
+namespace TerminalAppLocalTests
+{
+    class SettingsTests;
+}
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    struct TerminalSettings : TerminalSettingsT<TerminalSettings>, IInheritable<TerminalSettings>
+    {
+        TerminalSettings() = default;
+        TerminalSettings(const Model::CascadiaSettings& appSettings,
+                         guid profileGuid,
+                         const TerminalControl::IKeyBindings& keybindings);
+
+        static Model::TerminalSettings BuildSettings(const Model::CascadiaSettings& appSettings,
+                                                     const Model::NewTerminalArgs& newTerminalArgs,
+                                                     const TerminalControl::IKeyBindings& keybindings);

```

Comment:
Yes. This should be a constructor. Unfortunately, WinRT doesn't let us have two constructors with the same number of parameters. So I had to make `BuildSettings` static.

Reply:
I also had to break this up. To my knowledge, I'm not allowed to return a pair. So I'm basically just treating `BuildSettings` as a ctor.

Reply:
Changing these two functions to `CreateWithProfileByID` and `CreateWithNewTerminalArgs` as suggested by Dustin. So that basically fixes this. Resolving.